### PR TITLE
fix(antiddos): status returned in Create/Update/Delete func in flexibleengine_antiddos_v1 resource

### PR DESCRIPTION
**Pull request by azrod**
In crossplane the provider use `Read` Func before try create resource.
The terraform read function not catch err correctly therefore the
crossplane provider return err and do not try to create resource.

Error returned by crossplane :

```
Error retrieving AntiDdos: Action forbidden: [GET https://antiddos.eu-west-0.prod-cloud-ocb.orange-business.com/v1/xxx/antiddos/xxx], error message: {"error_code":"10001020", "error_msg":"IPID is invalid"}
```

In the fix use two method for catch notConfig Resource :
  * In Read func call checkNotConfig func which verifies if error is not in listed codes (10001020 and 10001020)
  * In Create/Update func edit loop stateChangeConf using
    `antiddos.GetStatus` instead `antiddos.Get` to determine status of
antiddos

Details of codes : 

- **10000016** - VPC cannot be accessed or the EIP does not exist.
- **10001020** - ID of the IP address is invalid.

Result of tests :

```
make testacc TEST='./flexibleengine' TESTARGS='-run TestAccAntiDdosV1_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./flexibleengine -v -run TestAccAntiDdosV1_basic -timeout 720m
=== RUN   TestAccAntiDdosV1_basic
--- PASS: TestAccAntiDdosV1_basic (219.31s)
PASS
ok      github.com/FlexibleEngineCloud/terraform-provider-flexibleengine/flexibleengine 219.852s
```

### CODE REVIEW

File: flexibleengine/utils.go [hunk 1]
```diff
@@ -21,12 +21,28 @@ type ErrorResp struct {
...
 }
 
-// ParseErrorMsg is used to unmarshal the error body to ErrorResp
-// usage: resp, pErr := ParseErrorMsg(err.Body)
-func ParseErrorMsg(body []byte) (ErrorResp, error) {
-	resp := ErrorResp{}
-	err := json.Unmarshal(body, &resp)
-	return resp, err
+// ParseErrorMap parses the error message from the API and returns a map
+func ParseErrorMsg(data interface{}) (ErrorResp, error) {
```
Comment (ShiChangkuo) [inline flexibleengine/utils.go:25]: @azrod the function is tested well, please keep it as is.

Comment (azrod) [reply flexibleengine/utils.go:25]: I'm getting an error and not a []byte. If you know how to do I let you modify

Comment (ShiChangkuo) [reply flexibleengine/utils.go:25]: you can use it like this: `ParseErrorMsg(err.Body)`

Comment (azrod) [reply flexibleengine/utils.go:25]:
```
errResp, nErr := ParseErrorMsg(err.Body)
```

`err.Body undefined (type error has no field or method Body)`

Comment (ShiChangkuo) [reply flexibleengine/utils.go:25]: you should convert the error to golangsdk.ErrDefault4xx

File: flexibleengine/resource_flexibleengine_antiddos_v1.go [hunk 2]
```diff
@@ -193,44 +198,40 @@ func resourceAntiDdosV1Delete(d *schema.ResourceData, meta interface{}) error {
...
 
-		r, err := antiddos.Get(antiddosClient, antiddosId).Extract()
-		if err != nil {
-			if _, ok := err.(golangsdk.ErrDefault403); ok {
-				log.Printf("[INFO] Successfully deleted antiddos %s", antiddosId)
-				return r, "notConfig", nil
-			}
-			return r, "normal", err
-		}
+	errResp, nErr := ParseErrorMsg(err.(golangsdk.ErrDefault400).Body)
```
Comment (ShiChangkuo) [inline flexibleengine/resource_flexibleengine_antiddos_v1.go:219]:
```suggestion
        if sdkErr, ok := err.(golangsdk.ErrDefault403); ok {
	        errResp, nErr := ParseErrorMsg(sdkErr.Body)
```

`comma ok` is recommended

Comment (azrod) [reply flexibleengine/resource_flexibleengine_antiddos_v1.go:219]: How to know the HTTP return code of the API?
400/403?

Comment (ShiChangkuo) [reply flexibleengine/resource_flexibleengine_antiddos_v1.go:219]: you can get more details from https://github.com/chnsz/golangsdk/blob/master/provider_client.go#L366

Comment (azrod) [reply flexibleengine/resource_flexibleengine_antiddos_v1.go:219]: Thanks but how to know the HTTP return code of errors 10000016 and 10001020 in AntiDDos API ?

Comment (ShiChangkuo) [reply flexibleengine/resource_flexibleengine_antiddos_v1.go:219]: `errResp.ErrorCode` is the error code

Comment (azrod) [reply flexibleengine/resource_flexibleengine_antiddos_v1.go]: Done

```
make testacc TEST='./flexibleengine' TESTARGS='-run TestAccAntiDdosV1_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./flexibleengine -v -run TestAccAntiDdosV1_basic -timeout 720m
=== RUN   TestAccAntiDdosV1_basic
--- PASS: TestAccAntiDdosV1_basic (221.36s)
PASS
ok      github.com/FlexibleEngineCloud/terraform-provider-flexibleengine/flexibleengine 222.011s
```

**Pull request merged**